rama-tcp/src/client/connect.rs

```diff
@@ -210,37 +207,41 @@ where
     let connected = Arc::new(AtomicBool::new(false));
     let sem = Arc::new(Semaphore::new(3));
 
-    // IPv6
-    let ipv6_tx = tx.clone();
-    let ipv6_domain = domain.clone();
-    let ipv6_connected = connected.clone();
-    let ipv6_sem = sem.clone();
-    ctx.spawn(tcp_connect_inner_branch(
-        dns.clone(),
-        connector.clone(),
-        IpKind::Ipv6,
-        ipv6_domain,
-        port,
-        ipv6_tx,
-        ipv6_connected,
-        ipv6_sem,
-    ));
-
-    // IPv4
-    let ipv4_tx = tx;
-    let ipv4_domain = domain.clone();
-    let ipv4_connected = connected.clone();
-    let ipv4_sem = sem;
-    ctx.spawn(tcp_connect_inner_branch(
-        dns,
-        connector,
-        IpKind::Ipv4,
-        ipv4_domain,
-        port,
-        ipv4_tx,
-        ipv4_connected,
-        ipv4_sem,
-    ));
+    match dns.mode {
+        DnsResolveIpMode::SingleIpV4 | DnsResolveIpMode::DualPreferIpV4 | DnsResovleIpMode::Dual =>{
```

Comment:
You probably want to provide utility methods to `DnsResolveIpMode`, such as:

```
fn ipv4_supported(&self) -> bool;
fn ipv6_supported(&self) -> bool;
```

Is same-same, but easier to have to update such statements only in a single loc.